### PR TITLE
fix(47820): Assertion crash when requesting refactorings in JSDoc comment with {@link } on next line

### LINKED ISSUE #47820: Assertion crash when requesting refactorings in JSDoc comment with `{@link }` on next line
Maybe related to https://github.com/Microsoft/TypeScript/issues/20809

```ts
/**
 * $
 * {@link Whatever}
 */
```

Request refactorings at `$`.

```
False expression: This failure could trigger https://github.com/Microsoft/TypeScript/issues/20809
Error: Debug Failure. False expression: This failure could trigger https://github.com/Microsoft/TypeScript/issues/20809
    at getErrorSpanForNode ([EXTENSION_DIR]\ms-vscode.vscode-typescript-next-4.6.20220208\node_modules\typescript\lib\tsserver.js:15006:22)
    at createDiagnosticForNodeInSourceFile ([EXTENSION_DIR]\ms-vscode.vscode-typescript-next-4.6.20220208\node_modules\typescript\lib\tsserver.js:14867:20)
    at Object.createDiagnosticForNode ([EXTENSION_DIR]\ms-vscode.vscode-typescript-next-4.6.20220208\node_modules\typescript\lib\tsserver.js:14858:16)
    at checkNode ([EXTENSION_DIR]\ms-vscode.vscode-typescript-next-4.6.20220208\node_modules\typescript\lib\tsserver.js:156726:36)
    at getRangeToExtract ([EXTENSION_DIR]\ms-vscode.vscode-typescript-next-4.6.20220208\node_modules\typescript\lib\tsserver.js:156644:53)
    at Object.getRefactorActionsToExtractSymbol [as getAvailableActions] ([EXTENSION_DIR]\ms-vscode.vscode-typescript-next-4.6.20220208\node_modules\typescript\lib\tsserver.js:156381:38)
    at [EXTENSION_DIR]\ms-vscode.vscode-typescript-next-4.6.20220208\node_modules\typescript\lib\tsserver.js:146883:30
    at getIterator ([EXTENSION_DIR]\ms-vscode.vscode-typescript-next-4.6.20220208\node_modules\typescript\lib\tsserver.js:562:23)
    at Object.next ([EXTENSION_DIR]\ms-vscode.vscode-typescript-next-4.6.20220208\node_modules\typescript\lib\tsserver.js:557:35)
    at Object.arrayFrom ([EXTENSION_DIR]\ms-vscode.vscode-typescript-next-4.6.20220208\node_modules\typescript\lib\tsserver.js:1309:88)
    at Object.getApplicableRefactors ([EXTENSION_DIR]\ms-vscode.vscode-typescript-next-4.6.20220208\node_modules\typescript\lib\tsserver.js:146879:23)
    at Object.getApplicableRefactors ([EXTENSION_DIR]\ms-vscode.vscode-typescript-next-4.6.20220208\node_modules\typescript\lib\tsserver.js:162405:32)
    at IpcIOSession.Session.getApplicableRefactors ([EXTENSION_DIR]\ms-vscode.vscode-typescript-next-4.6.20220208\node_modules\typescript\lib\tsserver.js:173454:53)
    at Session.handlers.ts.Map.ts.getEntries._a.<computed> ([EXTENSION_DIR]\ms-vscode.vscode-typescript-next-4.6.20220208\node_modules\typescript\lib\tsserver.js:172160:61)
    at [EXTENSION_DIR]\ms-vscode.vscode-typescript-next-4.6.20220208\node_modules\typescript\lib\tsserver.js:173822:88
    at IpcIOSession.Session.executeWithRequestId ([EXTENSION_DIR]\ms-vscode.vscode-typescript-next-4.6.20220208\node_modules\typescript\lib\tsserver.js:173813:28)
    at IpcIOSession.Session.executeCommand ([EXTENSION_DIR]\ms-vscode.vscode-typescript-next-4.6.20220208\node_modules\typescript\lib\tsserver.js:173822:33)
    at IpcIOSession.Session.onMessage ([EXTENSION_DIR]\ms-vscode.vscode-typescript-next-4.6.20220208\node_modules\typescript\lib\tsserver.js:173848:35)
    at process.<anonymous> ([EXTENSION_DIR]\ms-vscode.vscode-typescript-next-4.6.20220208\node_modules\typescript\lib\tsserver.js:176483:31)
    at process.emit (events.js:315:20)
    at emit (internal/child_process.js:903:12)
    at processTicksAndRejections (internal/process/task_queues.js:81:21)
```

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #47820

### CODE REVIEW

File: src/services/refactors/extractSymbol.ts [hunk 1]
```diff
@@ -299,7 +299,7 @@ namespace ts.refactor.extractSymbol {
             return { errors: [createFileDiagnostic(sourceFile, span.start, length, Messages.cannotExtractRange)] };
         }
 
-        if (isJSDoc(start)) {
```
Comment (gabritto) [inline src/services/refactors/extractSymbol.ts:302]: wow, a node test called `isJSDoc` that only checks if the node is a jsdoc *comment* seems very misleading... maybe we should rename it to `isJSDocComment`

**Review by gabritto — APPROVED** [review]
thanks ✨

File: src/services/refactors/extractSymbol.ts (hunk 1 above)
Comment (sandersn) [reply src/services/refactors/extractSymbol.ts:302]: https://github.com/microsoft/TypeScript/pull/44416

Comment (sandersn) [reply src/services/refactors/extractSymbol.ts:302]: (That's not precisely the same thing, but is definitely related)

**Review by sandersn — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
@typescript-bot cherry-pick this to release-4.6

**Pull request merged**